src/Crypt32.Shared/Crypt32+KeySpec.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    /// <content>
+    /// Contains the <see cref="KeySpec"/> nested type.
+    /// </content>
+    public partial class Crypt32
+    {
+        /// <summary>
+        /// An identifier that specifies the type of key.
+        /// </summary>
+        public enum KeySpec : uint
```

Comment:
LegacyKeySpec's constants are defined in wincrypt.h same as the CERT_NCRYPT_KEY_SPEC. However there is no legacy about CERT_NCRYPT_KEY_SPEC, so it would be wrong to extend LegacyKeySpec.
I think the best is to use uint instead of enum and define CERT_NCRYPT_KEY_SPEC as a public const. Except CryptAcquireCertificatePrivateKey no other API uses this constant.